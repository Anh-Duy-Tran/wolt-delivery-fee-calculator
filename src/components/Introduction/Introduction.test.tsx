import { Introduction } from '.';
import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';

describe('Introduction Component', () => {
  it('should render correct to the snapshot', () => {
    const { asFragment } = render(<Introduction />);
    expect(asFragment()).toMatchSnapshot();
  });
});
