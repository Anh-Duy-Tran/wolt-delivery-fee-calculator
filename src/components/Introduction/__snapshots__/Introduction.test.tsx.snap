// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Introduction Component > should render correct to the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col gap-4 w-full rounded-xl bg-white p-5 shadow-xl"
    data-testid="card"
  >
    <h2>
      INTRODUCTION
    </h2>
    <div
      class="flex items-center gap-2"
    >
      <div
        class="relative"
      >
        <div
          class="peer rounded ring-wolt-blue focus:ring-2"
          tabindex="0"
        >
          <a
            href="https://codecov.io/gh/Anh-Duy-Tran/wolt-delivery-fee-calculation"
          >
            <img
              src="https://codecov.io/gh/Anh-Duy-Tran/wolt-delivery-fee-calculation/graph/badge.svg?token=CJQV0NZ99N"
            />
          </a>
        </div>
        <div
          class="absolute right-0 z-50 mt-2 w-[350px] scale-75 rounded-md bg-slate-700 p-3 text-sm text-white opacity-0 transition peer-hover:scale-100 peer-hover:opacity-100 peer-focus:scale-100 peer-focus:opacity-100"
          data-testid="tooltip"
        >
          <p>
            Codecov Report: Shows the percentage of code covered by tests.
          </p>
        </div>
      </div>
      <div
        class="relative"
      >
        <div
          class="peer rounded ring-wolt-blue focus:ring-2"
          tabindex="0"
        >
          <a
            href="https://cloud.cypress.io/projects/jsr1iq/runs"
          >
            <img
              src="https://img.shields.io/endpoint?url=https://cloud.cypress.io/badge/simple/jsr1iq/main&style=flat&logo=cypress"
            />
          </a>
        </div>
        <div
          class="absolute right-0 z-50 mt-2 w-[350px] scale-75 rounded-md bg-slate-700 p-3 text-sm text-white opacity-0 transition peer-hover:scale-100 peer-hover:opacity-100 peer-focus:scale-100 peer-focus:opacity-100"
          data-testid="tooltip"
        >
          <p>
            Cypress Status: Indicates the results of automated end-to-end tests using Cypress.
          </p>
        </div>
      </div>
    </div>
    <div>
      <p>
        Integrated CI/CD pipelines with Codecov and Cypress badges display test results live, ensuring reliability.
      </p>
      <br />
      <img
        src="./intro-tooltip.gif"
      />
      <p
        class="text-center"
      >
        please try it out !
      </p>
      <br />
      <p>
        My project does more than just calculate fees - it demystifies them. Each fee component is clearly explained through interactive, reusable tooltips, offering users insight into their delivery costs with just a hover or click. I've also prioritized accessibility, making the app easily navigable and understandable for all users.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
